Minor bug fixes in getting parametric galaxies from COSMOSCatalog

arunkannawadi opened the pull request:
This is very short PR to fix a few bugs that occurs when one tries to get the parametric fits for galaxies from the recently updated COSMOS Catalog.

P.S.: Sorry, I am sleepy and have pushed the fix in existing branches :-/. It shouldn't bother with anything but let me know if it should be reverted.

arunkannawadi commented:
Disclaimer: I am fixing bugs as and when I see my script that was working fine earlier crashing. I haven't checked for other use cases.

rmandelb commented:
These look good to me.  Thanks for catching them.

Not sure what the best thing to do is about this appearing on other branches; if we merge it into master relatively soon given that it's pretty simple (I imagine someone else could also take a look pretty soon) then it won't cause any issue with other PRs.

arunkannawadi commented:
These are only about 4 lines. So if someone else as well, Mike or Josh,
could take a quick look at it, we can push it onto the master immediately.
On Apr 13, 2016 8:34 AM, "Rachel Mandelbaum" <notifications@github.com>
wrote:

> These look good to me. Thanks for catching them.
>
> Not sure what the best thing to do is about this appearing on other
> branches; if we merge it into master relatively soon given that it's pretty
> simple (I imagine someone else could also take a look pretty soon) then it
> won't cause any issue with other PRs.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/GalSim-developers/GalSim/pull/740#issuecomment-209408437>
>

jmeyers314 commented:
Looks good to me.  Can you add a short unit test though that would have caught these before you fixed them?

rmjarvis commented:
Agreed.  A could additional unit tests that would touch these lines of code would be helpful.

arunkannawadi commented:
I added some unit tests. I was having some trouble getting the Chromatic galaxies as my Pandas were throwing some error. I have Pandas v0.13.1, should I update it to be able to use it?

rmandelb commented:
What kind of errors?  With old versions, I was getting AttributeErrors, so the part of table.py that imports pandas looks like
```
try:
    import pandas
    ...other stuff...
except (ImportError, AttributeError):
   ...use np.loadtxt() instead...
```
So I think you can just add your type of exception to the list of exceptions that gets caught (and is used to decide not to use pandas).

rmandelb commented:
BTW I pushed a fix to address the way warnings were being given (the new code to address the case where there's only a parametric catalog was causing problems).

rmjarvis commented:
> So I think you can just add your type of exception to the list of exceptions that gets caught (and is used to decide not to use pandas).

I also ran across a machine which gave pandas errors.  I fixed this on master: b01b124c77f3f93519d0a090681a41daf6418142

rmjarvis commented:
All looks fine to me.  I also noticed a slight logic error here, with respect to the nobjects_only option, which I fixed.  I'm happy to merge this now.

rmandelb commented:
Thanks for adding the unit tests.  It looks good to me.  @arunkannawadi , please go ahead and merge as soon as you want.

arunkannawadi commented:
The pandas error seems to be resolved. But when I try to run 
```gal_param = cat.makeGalaxy(index=10,gal_type='parametric',chromatic=True)```,
 I get this error: 
```
  File "/usr/lib/python2.7/dist-packages/galsim/table.py", line 190, in __call__
    self.table.interpMany(x.astype(float),f)
AttributeError: '_LookupTable' object has no attribute 'interpMany'
```

rmandelb commented:
Are you saying you try to change the line in ```test_scene.py``` from 
```gal_param = cat.makeGalaxy(index=10,gal_type='parametric',chromatic=False)`` by changing the ```False``` to ```True```, and get an exception?  That runs just fine for me.  Have you definitely rerun ```scons``` and ```scons install``` on the latest version?

arunkannawadi commented:
Yes, I figured out that I haven't run scons recently, so I guess it's fine.
I will change the ```chromatic=False``` to ```chromatic=True``` for this
one case alone and merge with the master.

On Sun, Apr 17, 2016 at 10:51 AM, Rachel Mandelbaum <
notifications@github.com> wrote:

> Are you saying you try to change the line in test_scene.py from
> gal_param =
> cat.makeGalaxy(index=10,gal_type='parametric',chromatic=False)`` by
> changing theFalsetoTrue, and get an exception? That runs just fine for
> me. Have you definitely rerunsconsandscons install``` on the latest
> version?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GalSim-developers/GalSim/pull/740#issuecomment-211038031>
>



-- 
Regards

Arun Kannawadi Jayaraman
PhD candidate, Dept. of Physics,
Carnegie Mellon University,
Pittsburgh, PA 15213

rmandelb commented:
Well, no, if you do that then the rest of the test will fail.

rmandelb commented:
(Because it's not a GSObject, it's a ChromaticObject.)  I was saying that I don't get the same error as you, not that the test passes as-is.  The test needs to be changed to make sure it does the proper check for this one.

arunkannawadi commented:
Right. I fixed the complete test for Chromatic Object. I also fixed some
spacing issues with the error messages. Tests pass on my systems and I am
happy to merge now.

On Sun, Apr 17, 2016 at 11:28 AM, Rachel Mandelbaum <
notifications@github.com> wrote:

> (Because it's not a GSObject, it's a ChromaticObject.) I was saying that I
> don't get the same error as you, not that the test passes as-is. The test
> needs to be changed to make sure it does the proper check for this one.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GalSim-developers/GalSim/pull/740#issuecomment-211044088>
>



-- 
Regards

Arun Kannawadi Jayaraman
PhD candidate, Dept. of Physics,
Carnegie Mellon University,
Pittsburgh, PA 15213

rmjarvis commented on tests/test_scene.py:
```diff
@@ -53,6 +53,45 @@ def test_cosmos_basic():
+    gal_real = cat2.makeGalaxy(index=0,gal_type='real',chromatic=False)
+    if not isinstance(gal_real, galsim.RealGalaxy):
+        raise TypeError("COSMOS Catalog makeGalaxy routine does not return an instance of"+\
+                        "'galsim.RealGalaxy'")
```
You don't need the `\` here.  Or the `+` for that matter.  Python automatically concatenates two consecutive strings.

However, you should put a space at the end of the first string.  Otherwise you'll get "... of'galsim.RealGalaxy'".

arunkannawadi replied:
Ah, I didn't know that.

If there are no more comments, I'll merge it by tonight.
On Apr 17, 2016 4:55 PM, "Mike Jarvis" <notifications@github.com> wrote:

> In tests/test_scene.py
> <https://github.com/GalSim-developers/GalSim/pull/740#discussion_r59992520>
> :
>
> > @@ -53,6 +53,45 @@ def test_cosmos_basic():
> >      except ImportError:
> >          print 'The assert_raises tests require nose'
> >
> > +    # Try making galaxies
> > +    gal_real = cat2.makeGalaxy(index=0,gal_type='real',chromatic=False)
> > +    if not isinstance(gal_real, galsim.RealGalaxy):
> > +        raise TypeError("COSMOS Catalog makeGalaxy routine does not return an instance of"+\
> > +                        "'galsim.RealGalaxy'")
>
> You don't need the \ here. Or the + for that matter. Python automatically
> concatenates two consecutive strings.
>
> However, you should put a space at the end of the first string. Otherwise
> you'll get "... of'galsim.RealGalaxy'".
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GalSim-developers/GalSim/pull/740/files/1d814b617fb5b84dc1adbd77ad575717b84b51ca#r59992520>
>

arunkannawadi commented:
Merging.
